bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragment.java

```diff
@@ -131,16 +131,18 @@ public BookieSocketAddress getAddress(int bookieIndex) {
      * @return entryId
      */
     public long getFirstStoredEntryId() {
-        Long firstEntry = null;
+        Long firstEntry = LedgerHandle.INVALID_ENTRY_ID;
         for (int bookieIndex : bookieIndexes) {
             Long firstStoredEntryForBookie = getFirstStoredEntryId(bookieIndex);
-            if (null == firstEntry) {
-                firstEntry = firstStoredEntryForBookie;
-            } else if (null != firstStoredEntryForBookie) {
-                firstEntry = Math.min(firstEntry, firstStoredEntryForBookie);
+            if (firstStoredEntryForBookie != LedgerHandle.INVALID_ENTRY_ID) {
```

Comment:
this can lead to an NPE, if firstStoredEntryForBookie is null.

Reply:
`getFirstStoredEntryId(bookieIndex)` doesn't return null. It either returns valid 'firstEntry' or LedgerHandle.INVALID_ENTRY_ID. So we wouldn't run into this situation.